text/0000-delete-prefix.md

```diff
@@ -0,0 +1,135 @@
+- Feature Name: delete_by_prefix
+- Start Date: 2019-2-20
+- RFC PR:
+- Sawtooth Issue:
+
+# Summary
+[summary]: #summary
+
+This RFC proposes an additional API to transaction context,
+context.deletePrefix(address_prefix) which will get an address prefix as input
```

Comment:
Can we extend existing API and allow partial address?

Reply:
We could.
I thought in order to prevent TP developers from deleting prefix by mistake, the right approach will be to add a new API for this instead of adding capabilities to existing delete API.
Do you think it will be better to allow delete with prefix by extending existing API?

Reply:
Yes, this can however be controlled by outputs field in TransactionHeader.

Reply:
What would extending the existing API look like?

Reply:
@yoni-wolf please correct me, but I think that would look like the existing state delete call would take a list of address prefixes rather than a list of addresses. The validator would then need to distinguish each element of the list as between a full 70 character address as a single address or something less than 70 as a prefix.
Combining this: https://github.com/hyperledger/sawtooth-core/blob/master/protos/state_context.proto#L66-L70
With this: https://github.com/hyperledger/sawtooth-core/blob/master/protos/state_context.proto#L66-L70
@vaporos if that's not what you were asking please elaborate.

Reply:
Clarification was that this option should be included in the alternatives section.

Reply:
Added to the alternative section the option to use existing API instead of adding a new API